src/components/User/user.tsx

```diff
@@ -0,0 +1,198 @@
+import { Field, Form, Formik } from 'formik';
+import { useCallback, useState } from 'react';
+import { Button } from 'react-bootstrap';
+import '../../styles/user/user.css';
+
+type UserProps = Record<string, never>;
+
+type UserVariableRowProps = {
+  title: string;
+  content: string;
+  handleEditButtonClick: () => void;
+};
+
+type EditableUserVariableRowProps = {
+  title: string;
+  fieldNames: string[];
+  changeEditingStatus: () => void;
+  resetFields: () => void;
+  hideInput?: boolean;
+  requirePassword?: boolean;
+};
+
+const UserVariableRow: React.FC<UserVariableRowProps> = ({
+  content,
+  handleEditButtonClick,
+  title,
+}) => {
+  return (
+    <div className="flexRow">
+      <div className="flexColumn">
+        <span className="title">{title}</span>
+        <span className="content">{content}</span>
+      </div>
+      <Button
+        className="primaryButton"
+        onClick={handleEditButtonClick}
+        type="button"
+      >
+        Muokkaa
+      </Button>
+    </div>
+  );
+};
+
+const EditableUserVariableRow: React.FC<EditableUserVariableRowProps> = ({
+  fieldNames,
+  title,
+  changeEditingStatus,
+  resetFields,
+  requirePassword,
+  hideInput,
+}) => {
+  const handleCancelButtonClick = useCallback(() => {
+    changeEditingStatus();
+    resetFields();
+  }, [changeEditingStatus, resetFields]);
+
+  return (
+    <div className="flexRow">
+      <div className="flexColumn">
+        <span className="title">{title}</span>
+        {fieldNames.map((fieldName) => (
+          <Field
+            key={fieldName}
+            className="form-control"
+            name={fieldName}
+            type={hideInput ?? false ? 'password' : 'text'}
+          />
+        ))}
+
+        {requirePassword ?? false ? (
+          <>
+            <span className="title">Nykyinen salasana</span>
+            <Field type="password" className="form-control" name="password" />
+          </>
+        ) : null}
```

Comment:
Uusi ja vanha salasana menee nyt ikään kuin väärin päin. Tai ainakin eri päin kuin on normaalisti totuttu.

Reply:
Hmm, pitää pohtia tätä hieman koska nyt samaa komponenttia käytetään myös sähköpostin vaihtamisessa ja siinä tuntuisi oudolta kysyä salasanaa ennen sähköpostia

Reply:
Joo, sellanen tuttu flow olis
1) nykyinen salasana
2) uusi salasana
3) uusi salasana uudelleen